src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/UpdateAzureContainerRegistryCredential.cs

```diff
@@ -0,0 +1,58 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Management.ContainerRegistry.Models;
+
+namespace Microsoft.Azure.Commands.ContainerRegistry
+{
+    [Cmdlet(VerbsData.Update, ContainerRegistryCredentialNoun, SupportsShouldProcess = true),
+        OutputType(typeof(PSContainerRegistryCredential))]
+    public class UpdateAzureContainerRegistryCredential : ContainerRegistryCmdletBase
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Container Registry Name.")]
+        [Alias(ContainerRegistryNameAlias, RegistryNameAlias, ResourceNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The name of password to regenerate. " + AllowedPasswordNames)]
+        [ValidateNotNullOrEmpty]
+        public PasswordName PasswordName { get; set; }
+
```

Comment:
@djyou for all of your cmdlets (including this one), because you only need two parameters (`ResourceGroupName` and `Name`) to determine the registry container that you are performing an action on, using `ValueFromPipelineByPropertyName` for those two parameters instead of introducing a new parameter that will have the `ValueFromPipeline` attribute will definitely work. We try to push people towards passing objects into a cmdlet from the pipeline when they can, but in this case, because the number of parameters required is so small, it works just to use those instead of the object.